internal/app/init/main.go

```diff
@@ -188,13 +189,41 @@ func initram() (err error) {
 	return nil
 }
 
+func createOverlay(path string) error {
+	log.Printf("mounting overlay for %s\n", path)
+
+	parts := strings.Split(path, "/")
+	prefix := strings.Join(parts[1:], "-")
+	diff := fmt.Sprintf("/var/%s-diff", prefix)
+	workdir := fmt.Sprintf("/var/%s-workdir", prefix)
```

Comment:
Just making note for later, we should consider moving all of these to a specific dir/prefix ( ex, `/var/talos`, `/var/talos-overlays`, ... )